src/sdk/compilerfactory.cpp

```diff
@@ -277,9 +279,15 @@ void CompilerFactory::LoadSettings()
         event.SetClientData(static_cast<void*>(Compilers[i]));
         Manager::Get()->ProcessEvent(event);
 
-        if (Compilers[i]->GetMasterPath().IsEmpty())
-        {
-            Manager::Get()->GetLogManager()->DebugLog(F(_T("Master path of compiler ID \"%s\" is empty -> triggers auto-detection."), Compilers[i]->GetID().wx_str()));
+        if (    (Compilers[i]->GetMasterPath().IsEmpty() || !wxFileName::DirExists(Compilers[i]->GetMasterPath()))
+                &&
+                Compilers[i]->GetID().IsSameAs(defaultCompilerID)
+           )
+        {
+            if (Compilers[i]->GetMasterPath().IsEmpty())
+                Manager::Get()->GetLogManager()->DebugLog(F(_T("The master path of compiler ID \"%s\" is empty -> triggers auto-detection."), Compilers[i]->GetID().wx_str()));
+            if (!wxFileName::DirExists(Compilers[i]->GetMasterPath()))
+                Manager::Get()->GetLogManager()->DebugLog(F(_T("The master path (%s) of compiler ID \"%s\" does not exist -> triggers auto-detection."), Compilers[i]->GetMasterPath(), Compilers[i]->GetID().wx_str()));

```

Comment:
1. One common thing people like to do unfortunately - copy-paste repeated getters. Please store Compilers[i]->GetMasterPath() and Compilers[i]->GetID() in variables. Even repeating Compilers[i] on every line is a waste.
2. Please do not use F, use wxString::Format please.
3. Do not use wx_str() in format. Passing just wxStrings work.
4. Translate the messages you print in the log
5. Do not use _T or wxT. They are useless in wx3.x.

Reply:
This issue was hidden and I did not see it until tonight. I will need to make code changes to resolve these issues.